Week4/RecyclerView/app/src/main/java/com/janinculhaoglu/recyclerview/TVShowsAdapter.kt

```diff
@@ -0,0 +1,69 @@
+package com.janinculhaoglu.recyclerview
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.cell_tvshow.view.*
+
+
+class TVShowsAdapter(val clickListener: TvShowListClickListener) : RecyclerView.Adapter<TVShowsAdapter.TVShowsViewHolder>() {
+
+    interface TvShowListClickListener {
+        fun listItemClicked(list: TVShow)
+    }
+
+    fun onTVShowClick(index: Int, context: Context) {
+        val show = tvshowsList[index]
+        println(show.title) // Test to print the title of the selected cell
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TVShowsViewHolder {
+        // Create the view in XML
+        val rootView =
+            LayoutInflater.from(parent.context).inflate(R.layout.cell_tvshow, parent, false)
+        // Create a ViewHolder to control this view
+        val holder = TVShowsViewHolder(rootView)
+
+        return holder
+    }
+
+    override fun getItemCount(): Int {
+        return tvshowsList.size
+    }
+
+    override fun onBindViewHolder(holder: TVShowsViewHolder, position: Int) {
+        val tvShow = tvshowsList[position]
+        holder.fillWithTVShow(tvShow)
+        holder.itemView.setOnClickListener {
+            clickListener.listItemClicked(tvShow)
+        }
+    }
+
+    // Inner class to be able to have access to OnTVShowClick()
+    inner class TVShowsViewHolder(rootView: View) : RecyclerView.ViewHolder(rootView),
```

Comment:
I like to keep view holder class in a separate file. It makes code a bit cleaner and more separated, thus easier to maintain. 
This is an example how I would have created this:
```kotlin
class TVShowsViewHolder(override val containerView: View) :
        RecyclerView.ViewHolder(containerView), LayoutContainer {

        fun bind(tvShow: TVShow, clickListener: TvShowListClickListener) = with(containerView) {
            title.text = tvShow.title
            poster.setImageResource(tvShow.poster)
            genre.text = tvShow.genre
            rating.text = tvShow.rating.toString()
            rootView.setOnClickListener {
                clickListener.listItemClicked(tvShow)
            }
        }
    }
```